### PR TITLE
pin8, new recipe

### PR DESCRIPTION
**Pull request by cnelmortimer**
Dear repository masters,

Please find the HaikuPorts recipe for Pin8, a native Chip-8 emulator for Haiku OS written in C++.

Kind regards.

### CODE REVIEW

**Comment by diversys** [comment]
Please take a look at recipe formatting https://github.com/haikuports/haikuporter/blob/master/generic/generic_cmd-1.2.3.recipe
When you fix your recipe there is no need to close this PR and open new one. 
You can just force push to update this PR using: `git push -f`

File: games-emulation/pin8/pin8-1.0.recipe [hunk 1]
```diff
@@ -0,0 +1,49 @@
+SUMMARY="A Chip-8 emulator for Haiku OS"
+
+DESCRIPTION="This is a Chip-8 emulator for Haiku OS. Hence, it allows you to run Chip-8 games and software. Blitz, Blinky and Tetris are some games available for this platform. A sample game, a flappy bird clone, is also available in the webpage of this program. \ Pin8 features responsive design, real keyboard compatibility, sound and two running modes."
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: Wrap at 80th character. BTW, it's just Haiku, not Haiku OS

File: games-emulation/pin8/pin8-1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+HOMEPAGE="https://github.com/cnelmortimer/Pin8/blob/master/README.md"
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: https://github.com/cnelmortimer/Pin8

File: games-emulation/pin8/pin8-1.0.recipe [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+srcGitRev="5784af55e8e90d658533cce98492d3d13577460f"
+
+SOURCE_URI="https://github.com/cnelmortimer/Pin8/archive/$srcGitRev.tar.gz"
+
+CHECKSUM_SHA256="4997ae2d06a2346c37a8e9c9831e0d56f6132685b95054450d7f4c72e2ca6e00"
+
+SOURCE_DIR="Pin8-$srcGitRev"
+
+ARCHITECTURES="x86_gcc2 x86 ?x86_64"
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: We dropped x86 ARCH recently.

File: games-emulation/pin8/pin8-1.0.recipe [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+PROVIDES="
+	Pin8 = $portVersion
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: Lower case in this case

File: games-emulation/pin8/pin8-1.0.recipe [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+BUILD_REQUIRES="
+	haiku_devel
+	cmd:g++
+	cmd:make
+	cmd:xres
+	"
+	
+BUILD()
+{
+	make
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: Trailing whitespace

File: games-emulation/pin8/pin8-1.0.recipe [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+BUILD()
+{
+	make	
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
+	cp -a Pin8 $appsDir
+	addAppDeskbarSymlink $appsDir/Pin8
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: Trailing whitespace

**Comment by pulkomandy** [comment]
@diver: the x86 arch is still there, we just stopped distributing nightlies. It's still possible for people to build their own Haiku images if they wish so, and I don't see a reason for removing support yet?

**Comment by cnelmortimer** [comment]
Ok @diversys , I think I have done what you required (or at least, I really hope so... ). Sorry for any inconvenience and thank you for the guidelines.

File: games-emulation/pin8/pin8-1.0.recipe [hunk 7]
```diff
@@ -0,0 +1,43 @@
+SUMMARY="A Chip-8 emulator"
+DESCRIPTION="This is a Chip-8 emulator. Hence, it allows you to run Chip-8 \
+games  and software. Blitz, Blinky and Tetris are some games available for \
+this platform. A sample game, a flappy bird clone, is also available in the \
+webpage of this program. \Pin8 features responsive design, real keyboard \
+compatibility, sound and two running modes."
+HOMEPAGE="https://github.com/cnelmortimer/Pin8/blob/master/README.md"
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: Should be HOMEPAGE="https://github.com/cnelmortimer/Pin8"

File: games-emulation/pin8/pin8-1.0.recipe [hunk 8]
```diff
@@ -0,0 +1,43 @@
+SUMMARY="A Chip-8 emulator"
+DESCRIPTION="This is a Chip-8 emulator. Hence, it allows you to run Chip-8 \
+games  and software. Blitz, Blinky and Tetris are some games available for \
+this platform. A sample game, a flappy bird clone, is also available in the \
+webpage of this program. \Pin8 features responsive design, real keyboard \
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: Extra slash? \Pin8

File: games-emulation/pin8/pin8-1.0.recipe [hunk 9]
```diff
@@ -0,0 +1,43 @@
+SUMMARY="A Chip-8 emulator"
+DESCRIPTION="This is a Chip-8 emulator. Hence, it allows you to run Chip-8 \
+games  and software. Blitz, Blinky and Tetris are some games available for \
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: extra space "games  and"

File: games-emulation/pin8/pin8-1.0.recipe [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+srcGitRev="5784af55e8e90d658533cce98492d3d13577460f"
+SOURCE_URI="https://github.com/cnelmortimer/Pin8/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="4997ae2d06a2346c37a8e9c9831e0d56f6132685b95054450d7f4c72e2ca6e00"
+SOURCE_DIR="Pin8-$srcGitRev"
+
+ARCHITECTURES="x86_gcc2 ?x86_64"
+
+PROVIDES="
+	pin8 = $portVersion
+	app:pin8 = $portVersion
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: app:Pin8 should be uppercased (unlike the previous line which is needs to be always lowercase)

File: games-emulation/pin8/pin8-1.0.recipe [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	cmd:g++
+	cmd:make
+	cmd:xres
```
Comment (diversys) [inline games-emulation/pin8/pin8-1.0.recipe]: cmd:xres is provided by haiku package, I think it can be dropped here.

Comment (pulkomandy) [reply games-emulation/pin8/pin8-1.0.recipe]: in fact it is a good idea to declare it. If we ever decide to split the haiku package in smaller parts, or move the command to another package, we don't need to update any recipes then. That's the point of all these cmd: and lib: provides, and I see no reason to make an exception for the haiku package.

Comment (diversys) [reply games-emulation/pin8/pin8-1.0.recipe]: Noted

**Comment by cnelmortimer** [comment]
Hi again @diversys , 

I have removed the extra space after game, the extra slash before Pin8, cropped the homepage, uppercased "app:Pin8" and removed the "cmd:xres" dependency. Many thanks for your patience.
Kind regards.

**Pull request merged**